server/game/core/card/baseClasses/PlayableOrDeployableCard.ts

```diff
@@ -60,4 +80,68 @@ export class PlayableOrDeployableCard extends Card {
     protected setExhaustEnabled(enabledStatus: boolean) {
         this._exhausted = enabledStatus ? true : null;
     }
+
+    /** Create constant ability props on the card that decreases its cost under the given condition */
+    protected generateDecreaseCostAbilityProps(properties: IDecreaseEventCostAbilityProps<this>): IConstantAbilityProps {
+        const { title, condition, ...otherProps } = properties;
+
+        const costAdjusterProps: ICostAdjusterProperties = {
+            ...this.buildCostAdjusterGenericProperties(),
+            costAdjustType: CostAdjustType.Decrease,
+            ...otherProps
+        };
+
+        const effect = AbilityHelper.ongoingEffects.decreaseCost(costAdjusterProps);
+        return this.buildCostAdjusterAbilityProps(condition, title, effect);
+    }
+
+
+    /** Create constant ability props on the card that decreases its cost under the given condition */
+    protected generateIgnoreAllAspectPenaltiesAbilityProps(properties: IIgnoreAllAspectPenaltiesProps<this>): IConstantAbilityProps {
+        const { title, condition, ...otherProps } = properties;
+
+        const costAdjusterProps: IIgnoreAllAspectsCostAdjusterProperties = {
```

Comment:
nit, readability: could we make this ICostAdjusterProperties to line up with the other methods (or vice versa)?